core-bundle/src/Twig/Loader/ContaoFilesystemLoader.php

```diff
@@ -251,7 +251,7 @@ public function exists($name): bool
             return true;
         }
 
-        if (null !== ($themeTemplate = $this->getThemeTemplateName($name))) {
+        if (null !== ($themeTemplate = $this->getThemeTemplateName((string) $name))) {
```

Comment:
I am not sure about this at all. According to the phpDoc, `$name` cannot be anything but a string, so why does the underlying error occur in the first place?

Reply:
I think it's a Symfony bug that cannot really be fixed without breaking BC…

https://github.com/symfony/symfony/pull/26243#issuecomment-367304955
https://github.com/symfony/symfony/issues/25712
https://github.com/symfony/symfony/pull/15970

I could not find the parts where the old `TemplateReference` is used, yet, but according to @bytehead there seem quite a few Symfony bundles affected.

Reply:
And yes: I fully agree that this is a lot of nonsense. The interface should be the only thing that matters.

Reply:
Can we explicitly ignore non-string inputs?

```php
if (!\is_string($name)) {
    return false;
}
```

Reply:
IMO we could, yes.

Reply:
```suggestion
        if (\is_string($name) && null !== ($themeTemplate = $this->getThemeTemplateName( $name))) {
```

Reply:
This check should probably go in L250.

Reply:
IMHO we should not make assumptions for the parent class and only check the type in line 255.

Reply:
Our loader will probably never yield anything but `false` if a `TemplateReference` was input. :thinking: 

Btw.: if we go down this route, we also need to adjust `getSourceContext()`, `getCacheKey()` and `isFresh()`. They all call `normalizeName()` in the original `FilesystemLoader` that has a `(string)` cast.

Reply:
Added in b1bbd59cf.